src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -19,15 +29,28 @@ public class FindCommandParser implements Parser<FindCommand> {
      * @throws ParseException if the user input does not conform the expected format
      */
     public FindCommand parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
-        if (trimmedArgs.isEmpty()) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL,
+                PREFIX_STATUS, PREFIX_TAG);
+        if (!argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
-
-        String[] nameKeywords = trimmedArgs.split("\\s+");
-
-        return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL,
+                PREFIX_STATUS, PREFIX_TAG);
+
+        String[] nameKeyWords = argMultimap.getValue(PREFIX_NAME).orElse("").split("\\s+");
+        String[] phoneValues = argMultimap.getValue(PREFIX_PHONE).orElse("").split("\\s+");
+        String[] emailKeyWords = argMultimap.getValue(PREFIX_EMAIL).orElse("").split("\\s+");
+        String[] statusKeyWords = argMultimap.getValue(PREFIX_STATUS).orElse("").split("\\s+");
+        String[] tagKeyWords = argMultimap.getValue(PREFIX_TAG).orElse("").split("\\s+");
+
+        GeneralPersonPredicate generalPersonPredicate = new GeneralPersonPredicate(
```

Comment:
Yo, I think that maybe we should just give the arguments as the String[] instead of creating the predicate classes here.
This is to reduce the coupling so that only GeneralPersonPredicate has a dependency on the rest of Model, leaving the parser only dependent on the GeneralPersonPredicate.

I changed mine in the most my most recent PR if you want to see.

Reply:
hm okay i will make the changes